oauth_provider/tests/auth.py

```diff
@@ -57,24 +61,25 @@ def _request_token(self, method=METHOD_URL_QUERY, **parameters_overriden):
         elif method==METHOD_URL_QUERY:
             response = self.c.get("/oauth/request_token/", parameters)
         elif method==METHOD_POST_REQUEST_BODY:
-            body = urllib.urlencode(parameters)
+            body = six.moves.urllib.parse.urlencode(parameters)
             response = self.c.post("/oauth/request_token/", body, content_type="application/x-www-form-urlencoded")
         else:
             raise NotImplementedError
 
         if response.status_code != 200:
-            print response
+            print(response)
         self.assertEqual(response.status_code, 200)
 
-        response_qs = parse_qs(response.content)
-        self.assert_(response_qs['oauth_token_secret'])
-        self.assert_(response_qs['oauth_token'])
+        response_qs = parse_qs(response.content.decode('utf-8'))
+
+        self.assertTrue(response_qs['oauth_token_secret'])
+        self.assertTrue(response_qs['oauth_token'])
         self.assertEqual(response_qs['oauth_callback_confirmed'], ['true'])
 
         token = self.request_token = list(Token.objects.all())[-1]
-        self.assert_(token.key in response.content)
-        self.assert_(token.secret in response.content)
-        self.assert_(not self.request_token.is_approved)
+        self.assertTrue(token.key in response.content.decode('utf-8'))
```

Comment:
do we need to do this ?

Reply:
the `decode('utf-8')` converts `bytes` response to `string` so that the assertion is correct for `python3`